File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

//method prototypes
int letterCount(string text);
int wordCount(string text);
int sentenceCount(string text);
void printIndex(int index);

int main(int argc,string argv[])
{
    //prompting user input
    string text = get_string("Text: ");

    //calculating the grade index using formula
    float letterAvg = (float)letterCount(text) / (float)wordCount(text) * 100;
    float sentenceAvg= (float)(sentenceCount(text) || 1) / (float)wordCount(text) * 100;
    int index = round(0.0588 * letterAvg - 0.296 * sentenceAvg - 15.8);

    //checking text grade index
    printIndex(index);
}

void printIndex(int index)
{
    if(index <= 0)
    {
        printf("Before Grade 1\n");
    }
    else if( index <= 16)
    {
        printf("Grade %i\n",index);
    }
    else
    {
        printf("Grade 16+\n");
    }
}
//counting letters in text
int letterCount(string text)
{
    float lCount = 0;

    for(int i = 0; i <strlen(text); i++)
    {
        if(isalpha(text[i]))
        {
            lCount++;
        }
    }
    return lCount;
}
//counting words in text
int wordCount(string text)
{
    float wCount = 0;

     for(int i = 0; i <strlen(text); i++)
     {
        if(isspace(text[i]))
        {
            wCount++;
        }
     }
     return wCount +1 ;
}
//counting sentences in text
int sentenceCount(string text)
{
    int sCount = 0;

    for(int i = 0; i < strlen(text); i++)
    {
        if(text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sCount++;
        }
    }
    return sCount ;
}

File: sentimental-readability/readability.py
from cs50 import get_string

def main():
    text = input("Text:")

    # calculating grade index using formula
    letterAvg = letterCount(text) / wordCount(text) * 100
    sentenceAvg = sentenceCount(text) / wordCount(text) * 100
    index = round(0.0588 * letterAvg - 0.296 * sentenceAvg - 15.8)

    printIndex(index)

#print index
def printIndex(index):
    if index <= 0:
        print("Before Grade 1")
    elif index <= 16:
        print("Grade {index}", index)
    else:
        print("Grade 16+")

#letter count method
def letterCount(text):
    lCount = 0

    for i in range(0,len(text)):
        if text[i].isalpha():
            lCount += 1
    return lCount

#word count method
def wordCount(text):
    wCount = 0

    for i in range(0,len(text)):
        if text[i] == " ":
            wCount +=1
    return wCount +1

# sentence count method
def sentenceCount(text):
    sCount = 0

    for i in range(0,len(text)):
        if text[i] == "." or text[i] == "?" or text[i] ==  "!":
            sCount += 1
    return sCount

# checks if script is run directly/imported.Call main
if __name__ == "__main__":
    main()
